core/services/relay/evm/chain_writer.go

```diff
@@ -30,11 +32,18 @@ type ChainWriterService interface {
 // Compile-time assertion that chainWriter implements the ChainWriterService interface.
 var _ ChainWriterService = (*chainWriter)(nil)
 
-func NewChainWriterService(logger logger.Logger, client evmclient.Client, txm evmtxmgr.TxManager, config types.ChainWriterConfig) (ChainWriterService, error) {
+func NewChainWriterService(logger logger.Logger, client evmclient.Client, txm evmtxmgr.TxManager, estimator gas.EvmFeeEstimator, config types.ChainWriterConfig) (ChainWriterService, error) {
+	maxGasPrice := config.MaxGasPrice
+	if maxGasPrice == nil {
+		maxGasPrice = assets.GWei(500).ToInt()
```

Comment:
What is this value based on? We already have mechanisms for configuration defaults and overrides. I don't think it is a good idea to splinter configuration like this. It would be simple and clear to return an error if the value is unspecified.

Reply:
Updated to return an error